eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/CommandCodecs.scala

```diff
@@ -19,14 +19,53 @@ package fr.acinq.eclair.wire.internal
 import akka.actor.ActorRef
 import fr.acinq.eclair.channel._
 import fr.acinq.eclair.wire.protocol.CommonCodecs._
-import fr.acinq.eclair.wire.protocol.FailureMessageCodecs.failureMessageCodec
+import fr.acinq.eclair.wire.protocol.FailureMessageCodecs._
+import fr.acinq.eclair.wire.protocol.LightningMessageCodecs.{channelFlagsCodec, messageFlagsCodec}
+import fr.acinq.eclair.wire.protocol._
+import fr.acinq.eclair.{BlockHeight, MilliSatoshiLong}
 import scodec.Codec
 import scodec.codecs._
 
 import scala.concurrent.duration.FiniteDuration
 
 object CommandCodecs {
 
+  // A trailing tlv stream was added in https://github.com/lightning/bolts/pull/1021 which wasn't handled properly by
```

Comment:
Damn! How about we just use a generic hardcoded error message? We should be able to work around the absence of length delimitation, because trailing fields are constant-size.

Reply:
I'm not sure what you mean...we really need to introduce a new codec anyway that adds a length-prefix to failure messages, otherwise we won't be able to properly encode/decode failure messages that contain tlvs. I don't see how we could introduce that while keeping compatibility with the existing codec?

Another option could be to update the codec and do a DB migration. But that would also require having the legacy codec around (in the DB migration code), it doesn't remove it...

Reply:
Right, I didn't understand your earlier comment, that's a good idea to simplify the code!

Reply:
Done in https://github.com/ACINQ/eclair/pull/2455/commits/5caaadc4bd5a4efac8c18b63fd61bf1d60a4f54a